### PR TITLE
Sort dashboards based on pins

### LINKED ISSUE #5457: Order dashboards in overview by pin status
## Is your feature request related to a problem?

When going to the dashboard page, I'd like all the pinned dashboards to show at the top of the list.
## Describe the solution you'd like

On the dashboard overview page, there should be 2 subsections:
1. Pinned dashboards, sorted alphabetically
2. Unpinned dashboards, sorted alphabetically

## Describe alternatives you've considered
Leave as-is and refer to the left-nav when wanting to view dashboards in this manner. When you have a lot of dashboards it's tough to go through that list and I find myself clicking in most of the time.

## Additional context
It's sorted this way in the left nav hover menu - I think it would be nice to use this same type of ordering on the overview page as well.

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Fixes #5457 

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: frontend/src/scenes/dashboard/Dashboards.tsx [hunk 1]
```diff
@@ -42,6 +42,14 @@ export function Dashboards(): JSX.Element {
                     </span>
                 )
             },
+            sorter: {
+                multiple: 20,
+                compare: (a, b) => {
+                    const intValue = +a.pinned + +b.pinned
```
Comment (alexkim205) [inline frontend/src/scenes/dashboard/Dashboards.tsx:48]: I'm kind of confused, what does this +(boolean) syntax do?

**Comment by alexkim205** [comment]
Just spot tested it, but the sorting doesn't seem to be working

Screen Shot 2021-08-13 at 10 45 30 AM

**Review by alexkim205 — CHANGES_REQUESTED** [review]

**Comment by paolodamico** [comment]
Just cleaned up the code a bit to make it more clear, and tested again and seems to be working. Can you review again @alexkim205 please?

**Comment by paolodamico** [comment]
Ping @alexkim205 and/or @kpthatsme

**Comment by alexkim205** [comment]
Looks great!

**Review by alexkim205 — APPROVED** [review]

**Pull request merged**